lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,49 @@
+Facter.add(:powershell_installed) do
+ confine :operatingsystem => :windows
+ require 'win32/registry'
```

Comment:
This may require checking to see if we are in Windows. `if Puppet::Util::Platform.windows?`  - https://github.com/chocolatey/puppet-chocolatey/blob/master/lib/facter/choco_install_path.rb#L2-L19